trl/trainer/iterative_sft_trainer.py

```diff
@@ -0,0 +1,393 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from typing import List, Optional
+
+import torch
+from accelerate import Accelerator
+from accelerate.utils import ProjectConfiguration
+from datasets import Dataset
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from transformers import (
+    DataCollator,
+    DataCollatorForLanguageModeling,
+    DataCollatorForSeq2Seq,
+    PreTrainedModel,
+    PreTrainedTokenizerBase,
+)
+
+from ..core import PPODecorators, set_seed
+from ..import_utils import is_torch_greater_2_0
+from . import IterativeSFTConfig
+
+
+class IterativeSFTTrainer:
```

Comment:
I wonder if we should inherit from `transformers.Trainer` which would give us a lot of init logic as well as e.g. saving models or evaluating for free. 

As discussed with @lewtun offline, maybe we should inherit from the `Trainer` everywhere to have the same base logic for all trainers